cmd/util/cmd/epochs/cmd/deploy.go

```diff
@@ -256,14 +257,18 @@ func getDeployEpochTransactionText(snapshot *inmem.Snapshot) []byte {
 	chainID := head.ChainID
 	systemContracts := systemcontracts.SystemContractsForChain(chainID)
 
+	env := coreContractstemplates.Environment{
+		FungibleTokenAddress:     flagFungibleTokenAddress,
+		FlowTokenAddress:         flagFlowTokenAddress,
+		IDTableAddress:           flagIDTableAddress,
+		QuorumCertificateAddress: systemContracts.ClusterQC.Address.Hex(),
+		DkgAddress:               systemContracts.DKG.Address.Hex(),
+		FlowFeesAddress:          flagFlowFeesAddress,
+	}
```

Comment:
> That would require a circular dependency because core-contracts/lib/go/templates uses templates.Environment to replace the address placeholders, so I'd prefer not to do that

Not sure I  follow...

What I propose is that the `SystemContracts` receiver `func (c SystemContracts) AsTemplateEnv() templates.Environment` can be used here. It essentially already does exactly what you did below: sets all the addresses of the `templates.Environment` to the addresses from `SystemContracts`.

In this instance its only somewhat useful:

```suggestion
	env := systemContracts.AsTemplateEnv() // this fills all addresses
	// but in this case some need to be manually overriden (not actually sure why, we should know the location of all these contracts already)
	env.FungibleTokenAddress = flagFungibleTokenAddress
	env.FlowTokenAddress = flagFlowTokenAddress
	env.IDTableAddress = flagIDTableAddress
	env.FlowFeesAddress = flagFlowFeesAddress
```

Reply:
oh, I thought you meant change `core-contracts/lib/go/templates` to use it. I can update this like you suggest